src/ctia/http/server.clj

```diff
@@ -208,6 +209,9 @@
             (when-let [lifetime (:lifetime-in-sec jwt)]
               {:jwt-max-lifetime-in-sec lifetime}))))
 
+         ;; encode error responses from ring-jwt-middleware
+         (:enabled jwt) (wrap-restful-response (handler/->format-options))
```

Comment:
The `wrap-restful-response` middleware is already applied by [compojure-api](https://github.com/metosin/compojure-api/blob/1.1.13/src/compojure/api/middleware.clj#L244). This middleware is not explicitly defined as idempotent. We don't know how it behaves with all content types and we don't want to rely on its implementation.

We don't have this issue in IROH, the [`wrap-jwt-auth-fn`](https://github.com/threatgrid/iroh/blob/b89cfdd47b3ea5ab0c5664ae2a3182ae346be3a9/lib/iroh-web/src/iroh_web/core.clj#L291) is passed to the `compojure.api.api/api` fn. Could we have the same approach with CTIA? It was probably not possible before the integration of TK in CTIA.

Reply:
> This middleware is not explicitly defined as idempotent.

If the primary concern is idempotency, here's an alternative implementation that immediately wraps the `:error-handler` in `wrap-restful-response`, and so it does not get called repeatedly: https://github.com/threatgrid/ctia/compare/06bdb88373a4dbe1b362caac3a290fa7fb27acac...7c7b1db66db7d0f5b067964a3a69e583a4ffcf7b?w=1#diff-fd6005488e3bd6e64d13d9d694ad18bfea2ea0ed5be6f2a0d2d81794275c3e85L187-R233

FWIW `wrap-restful-response` is [documented](https://github.com/ngrunwald/ring-middleware-format) to be idempotent on string/binary `:body` (`If a route returns a String, File, InputStream or nil, nothing will be done.`). With that assumption, I settled on the current implementation.

> Could we have the same approach with CTIA?

It would be cleaner and I've tried it unsuccessfully several times. I think it would be safer for someone more familiar with the security side of things to try, since it may involve changing the order of some middleware.

Reply:
> FWIW wrap-restful-response is documented to be idempotent on string/binary :body (If a route returns a String, File, InputStream or nil, nothing will be done.). With that assumption, I settled on the current implementation.

Then the current implementation is correct. Please add a comment explaining why we can use `wrap-restful-response` here even it it's already included in the compojure API middlewares. 